android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestSuggestionProvider.kt

```diff
@@ -92,4 +105,21 @@ internal data class SuggestionDetails(
     val fullKeyword: String,
     val isSponsored: Boolean,
     val icon: List<UByte>?,
+    val clickInfo: FxSuggestClickInfo? = null,
 )
+
+/**
+ * Collective of fields required for fxsuggest click telemetry
+ */
+sealed interface FxSuggestClickInfo {
+    /**
+     * AMP related fields for fx suggest click telemetry
+     */
+    data class Amp(
+        val blockId: String,
```

Comment:
```suggestion
        val blockId: Int,
```

Question, minor: How would you feel about passing `blockId` as an integer here, then using `.toString()` when you call `FxSuggest.blockId.set`. (And the same question about converting the advertiser to lowercase here vs. in `MetricController`).

It's OK with me either way 😊 I'd love to hear your thoughts, though!